isis/tests/Fixtures.cpp

```diff
@@ -2,203 +2,68 @@
-    testIsd = json::parse(R"(
-      {"CameraVersion": 2,
-         "NaifKeywords": {
-              "BODY_CODE" : 499,
-              "BODY499_RADII" : [3396.19, 3396.19, 3376.2],
-              "BODY_FRAME_CODE" : 10014,
-              "CLOCK_ET_-27999_33322515_COMPUTED" : "d5b9203a4d24c5c1",
-              "INS-27002_TRANSX" : [0.0, 0.011764705882353, 0.0],
-              "INS-27002_TRANSY" : [0.0, 0.0, 0.01176470588235],
-              "INS-27002_ITRANSS" : [0.0, 85.0, 0.0],
-              "INS-27002_ITRANSL" : [0.0, 0.0, 85.0]
-        },
-      "InstrumentPointing": {
-        "TimeDependentFrames": [-85600, -85000, 1],
-        "CkTableStartTime": 100,
-        "CkTableEndTime": 100.1,
-        "CkTableOriginalSize": 2,
-        "EphemerisTimes": [
-          100,
-          100.1
-        ],
-        "Quaternions": [
-          [0.0, -0.660435174378928, 0, 0.750883067090392],
-          [0.0, -0.660435174378928, 0, 0.750883067090392]
-        ],
-        "AngularVelocity": [
-          [0, 0, 0],
-          [0, 0, 0]
-        ],
-        "ConstantFrames": [-85600],
-        "ConstantRotation": [1, 0, 0, 0, 1, 0, 0, 0, 1]
-      },
-      "BodyRotation": {
-        "TimeDependentFrames": [31006, 1],
-        "CkTableStartTime": 100,
-        "CkTableEndTime": 100.1,
-        "CkTableOriginalSize": 2,
-        "EphemerisTimes": [
-          100,
-          100.1
-        ],
-        "Quaternions": [
-          [ 0, 0.8509035, 0, 0.525322 ],
-          [ 0, 0.8509035, 0, 0.525322 ]
-        ],
-        "AngularVelocity": [
-          [0, 0, 0],
-          [0, 0, 0]
-        ],
-        "ConstantFrames": [31001, 31007, 31006],
-        "ConstantRotation": [-0.4480736,  0,  0.8939967, 0,  1,  0, -0.8939967,  0, -0.4480736]
-      },
-      "InstrumentPosition": {
-        "SpkTableStartTime": 100,
-        "SpkTableEndTime": 100.1,
-        "SpkTableOriginalSize": 2,
-        "EphemerisTimes": [
-          100,
-          100.1
-        ],
-        "Positions": [
-          [1000, 0, 0],
-          [1000, 0, 0]
-        ],
-        "Velocities": [
-          [0, 0, 0],
-          [0, 0, 0]
-        ]
-      },
-      "SunPosition": {
-        "SpkTableStartTime": 100,
-        "SpkTableEndTime": 100.1,
-        "SpkTableOriginalSize": 2,
-        "EphemerisTimes": [
-          100,
-          100.1
-        ],
-        "Positions": [
-          [0, 20, 0]
-        ],
-        "Velocities": [
-          [10,10,10]
-        ]
-      }
-    })");
-
-    // call constructor
-    Cube testCube;
+  void DefaultCube::SetUp() {
+    std::ifstream isdFile("data/defaultImage/defaultCube.isd");
+    std::ifstream cubeLabel("data/defaultImage/defaultCube.pvl");
     
-    testCube.fromIsd(tempFile.fileName() + ".cub", label, testIsd, "rw");
+    isdFile >> isd;
+    cubeLabel >> label;
+
+    testCube->fromIsd(tempFile.fileName() + ".cub", label, isd, "rw");
 
     // reopen to write out new data 
-    testCube.reopen("rw");
+    testCube->reopen("rw");
```

Comment:
Now that I'm looking at this again, I'm confused by why `fromIsd` closes and reopens the cube in the function and then it needs to be reopened again here? Does leaving this out break it?

Reply:
It should be fine, seems like a missed detail when converting the fixture logic over to fromIsd. reopen was added to fromIsd but wasn't removed from the fixtures.